agent/src/beerocks/slave/link_metrics/link_metrics.h

```diff
@@ -0,0 +1,123 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2020 MaxLinear
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+#ifndef __LINK_METRICS_H__
+#define __LINK_METRICS_H__
+
+#include <string>
+
+#include <tlvf/common/sMacAddr.h>
+
+namespace beerocks {
+
+/**
+ * @brief Transmitter metrics information associated to the link between a local interface and a
+ * neighbor's interface.
+ *
+ * Information in this structure is used to fill Transmitter Link Metric TLV.
+ */
+struct sTransmitterLinkMetrics {
+    /**
+     * Estimated number of lost packets on the transmit side of the link during the measurement
+     * period.
+     */
+    uint32_t packet_errors = 0;
+
+    /**
+     * Estimated number of packets transmitted by the Transmitter of the link on the same
+     * measurement period used to estimate tx_packet_errors.
+     */
+    uint32_t transmitted_packets = 0;
+
+    /**
+     * The maximum MAC throughput of the link estimated at the transmitter and expressed in Mb/s.
+     */
+    uint32_t mac_throughput_capacity_mbps = 0;
+
+    /**
+     * The estimated average percentage of time that the link is available for data transmissions.
+     */
+    uint16_t link_availability = 0;
+
+    /**
+     * If the media type of the link is IEEE 802.3, IEEE 1901, MoCA 1.1 or Generic Phy, then this
+     * value is the PHY rate estimated at the transmitter of the link expressed in Mb/s;
+     * otherwise, it is set to 0xFFFF.
+     */
+    uint32_t phy_rate_mbps = 0;
+};
+
+/**
+ * @brief Receiver metrics information associated to the link between a local interface and a
+ * neighbor's interface.
+ *
+ * Information in this structure is used to fill Receiver Link Metric TLV.
+ */
+struct sReceiverLinkMetrics {
+    /**
+     * Estimated number of lost packets during the measurement period.
+     */
+    uint32_t packet_errors = 0;
+
+    /**
+     * Number of packets received at the interface during the same measurement period used to count
+     * packet_errors
+     */
+    uint32_t packets_received = 0;
+
+    /**
+     * If the media type of the link is IEEE 802.11, then this value is the estimated RSSI in dB at
+     * the receive side of the link expressed in dB; otherwise, it is set to 0xFF.
+     */
+    uint8_t rssi = 0;
```

Comment:
So why not set it to UINT8_MAX by default instead of 0? Then you don't need to set it in the ethernet link collector implementation.

Reply:
This struct is going to be used in the WiFi link collector implementation too. UINT8_MAX is not a default value, but the value to use for ethernet links (according to standard)